src/Tasks/Copy.cs

```diff
@@ -323,29 +349,51 @@ CopyFileWithState copyFile
                 return false;
             }
 
-            bool success = true;
-
             // Environment variable stomps on user-requested value if it's set. 
-            if (Environment.GetEnvironmentVariable("MSBUILDALWAYSOVERWRITEREADONLYFILES") != null)
+            if (Environment.GetEnvironmentVariable(AlwaysOverwriteReadOnlyFilesEnvVar) != null)
             {
                 OverwriteReadOnlyFiles = true;
             }
 
-            // Build up the sucessfully copied subset
-            var destinationFilesSuccessfullyCopied = new List<ITaskItem>();
+            // Track successfully copied subset.
+            List <ITaskItem> destinationFilesSuccessfullyCopied;
+            bool success = parallelism == 1
+                ? CopySingleThreaded(copyFile, out destinationFilesSuccessfullyCopied)
```

Comment:
Always use CopySingleThreaded when we only copy a single file.

Reply:
Done in next iteration